[2024-12-08T14:00:21.441+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-08T14:00:21.461+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sparking_flow.python_job scheduled__2024-12-06T00:00:00+00:00 [queued]>
[2024-12-08T14:00:21.469+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sparking_flow.python_job scheduled__2024-12-06T00:00:00+00:00 [queued]>
[2024-12-08T14:00:21.470+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2024-12-08T14:00:21.484+0000] {taskinstance.py:2889} INFO - Executing <Task(SparkSubmitOperator): python_job> on 2024-12-06 00:00:00+00:00
[2024-12-08T14:00:21.498+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=193) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-12-08T14:00:21.496+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'sparking_flow', 'python_job', 'scheduled__2024-12-06T00:00:00+00:00', '--job-id', '135', '--raw', '--subdir', 'DAGS_FOLDER/spark_***.py', '--cfg-path', '/tmp/tmp94630k1q']
[2024-12-08T14:00:21.499+0000] {standard_task_runner.py:72} INFO - Started process 197 to run task
[2024-12-08T14:00:21.500+0000] {standard_task_runner.py:105} INFO - Job 135: Subtask python_job
[2024-12-08T14:00:21.590+0000] {task_command.py:467} INFO - Running <TaskInstance: sparking_flow.python_job scheduled__2024-12-06T00:00:00+00:00 [running]> on host 2a892a3469ec
[2024-12-08T14:00:21.708+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Yusuf Ganiyu' AIRFLOW_CTX_DAG_ID='sparking_flow' AIRFLOW_CTX_TASK_ID='python_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-06T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-12-06T00:00:00+00:00'
[2024-12-08T14:00:21.709+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-12-08T14:00:21.737+0000] {base.py:84} INFO - Retrieving connection 'spark-conn'
[2024-12-08T14:00:21.739+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py
[2024-12-08T14:00:30.245+0000] {spark_submit.py:634} INFO - file copied
[2024-12-08T14:00:30.850+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:30 INFO SparkContext: Running Spark version 3.5.3
[2024-12-08T14:00:30.857+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:30 INFO SparkContext: OS info Linux, 6.10.14-linuxkit, aarch64
[2024-12-08T14:00:30.865+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:30 INFO SparkContext: Java version 17.0.13
[2024-12-08T14:00:31.032+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:31 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-12-08T14:00:31.316+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:31 INFO ResourceUtils: ==============================================================
[2024-12-08T14:00:31.318+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:31 INFO ResourceUtils: No custom resources configured for spark.driver.
[2024-12-08T14:00:31.321+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:31 INFO ResourceUtils: ==============================================================
[2024-12-08T14:00:31.322+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:31 INFO SparkContext: Submitted application: PythonWordCount
[2024-12-08T14:00:31.375+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:31 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2024-12-08T14:00:31.389+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:31 INFO ResourceProfile: Limiting resource is cpu
[2024-12-08T14:00:31.391+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:31 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2024-12-08T14:00:31.530+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:31 INFO SecurityManager: Changing view acls to: ***
[2024-12-08T14:00:31.534+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:31 INFO SecurityManager: Changing modify acls to: ***
[2024-12-08T14:00:31.541+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:31 INFO SecurityManager: Changing view acls groups to:
[2024-12-08T14:00:31.551+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:31 INFO SecurityManager: Changing modify acls groups to:
[2024-12-08T14:00:31.554+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:31 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2024-12-08T14:00:32.415+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:32 INFO Utils: Successfully started service 'sparkDriver' on port 44515.
[2024-12-08T14:00:32.521+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:32 INFO SparkEnv: Registering MapOutputTracker
[2024-12-08T14:00:32.675+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:32 INFO SparkEnv: Registering BlockManagerMaster
[2024-12-08T14:00:32.819+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:32 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2024-12-08T14:00:32.829+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:32 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2024-12-08T14:00:32.977+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:32 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2024-12-08T14:00:33.083+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:33 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-596cb091-17f8-4477-a57b-4e18a3c6486c
[2024-12-08T14:00:33.174+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:33 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2024-12-08T14:00:33.296+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:33 INFO SparkEnv: Registering OutputCommitCoordinator
[2024-12-08T14:00:34.053+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:34 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2024-12-08T14:00:34.292+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:34 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
[2024-12-08T14:00:34.295+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:34 WARN Utils: Service 'SparkUI' could not bind on port 4041. Attempting port 4042.
[2024-12-08T14:00:34.302+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:34 WARN Utils: Service 'SparkUI' could not bind on port 4042. Attempting port 4043.
[2024-12-08T14:00:34.303+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:34 WARN Utils: Service 'SparkUI' could not bind on port 4043. Attempting port 4044.
[2024-12-08T14:00:34.304+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:34 WARN Utils: Service 'SparkUI' could not bind on port 4044. Attempting port 4045.
[2024-12-08T14:00:34.313+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:34 INFO Utils: Successfully started service 'SparkUI' on port 4045.
[2024-12-08T14:00:34.793+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:34 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://sparkingflow-spark-master-1:7077...
[2024-12-08T14:00:35.072+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:35 INFO TransportClientFactory: Successfully created connection to sparkingflow-spark-master-1/172.18.0.5:7077 after 167 ms (0 ms spent in bootstraps)
[2024-12-08T14:00:35.839+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:35 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20241208140035-0003
[2024-12-08T14:00:35.901+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:35 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 33843.
[2024-12-08T14:00:35.904+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:35 INFO NettyBlockTransferService: Server created on 2a892a3469ec:33843
[2024-12-08T14:00:35.916+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:35 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2024-12-08T14:00:35.955+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:35 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 2a892a3469ec, 33843, None)
[2024-12-08T14:00:35.971+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:35 INFO BlockManagerMasterEndpoint: Registering block manager 2a892a3469ec:33843 with 434.4 MiB RAM, BlockManagerId(driver, 2a892a3469ec, 33843, None)
[2024-12-08T14:00:35.982+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:35 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 2a892a3469ec, 33843, None)
[2024-12-08T14:00:35.992+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:35 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 2a892a3469ec, 33843, None)
[2024-12-08T14:00:36.868+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:36 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2024-12-08T14:00:39.274+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:39 INFO SparkContext: Added file /tmp/test1234.csv at spark://2a892a3469ec:44515/files/test1234.csv with timestamp 1733666439267
[2024-12-08T14:00:39.295+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:39 INFO Utils: Copying /tmp/test1234.csv to /tmp/spark-1cd0e7b8-9745-43c8-8c3a-c004433dfa05/userFiles-288d223e-9a2c-48e5-acd4-691d23aa6c92/test1234.csv
[2024-12-08T14:00:39.332+0000] {spark_submit.py:634} INFO - myfile: /tmp/test1234.csv
[2024-12-08T14:00:39.334+0000] {spark_submit.py:634} INFO - Resolved file path: /tmp/spark-1cd0e7b8-9745-43c8-8c3a-c004433dfa05/userFiles-288d223e-9a2c-48e5-acd4-691d23aa6c92/test1234.csv
[2024-12-08T14:00:39.348+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:39 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2024-12-08T14:00:39.355+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:39 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2024-12-08T14:00:41.604+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:41 INFO InMemoryFileIndex: It took 82 ms to list leaf files for 1 paths.
[2024-12-08T14:00:41.699+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:41 INFO InMemoryFileIndex: It took 4 ms to list leaf files for 1 paths.
[2024-12-08T14:00:45.358+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:45 INFO FileSourceStrategy: Pushed Filters:
[2024-12-08T14:00:45.363+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:45 INFO FileSourceStrategy: Post-Scan Filters: (length(trim(value#0, None)) > 0)
[2024-12-08T14:00:46.224+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO CodeGenerator: Code generated in 389.856625 ms
[2024-12-08T14:00:46.289+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO MemoryStore: Block broadcast_0 stored as values in memory (estimated size 199.6 KiB, free 434.2 MiB)
[2024-12-08T14:00:46.364+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO MemoryStore: Block broadcast_0_piece0 stored as bytes in memory (estimated size 34.3 KiB, free 434.2 MiB)
[2024-12-08T14:00:46.367+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 2a892a3469ec:33843 (size: 34.3 KiB, free: 434.4 MiB)
[2024-12-08T14:00:46.373+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO SparkContext: Created broadcast 0 from load at NativeMethodAccessorImpl.java:0
[2024-12-08T14:00:46.385+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO FileSourceScanExec: Planning scan with bin packing, max size: 4194304 bytes, open cost is considered as scanning 4194304 bytes.
[2024-12-08T14:00:46.557+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO SparkContext: Starting job: load at NativeMethodAccessorImpl.java:0
[2024-12-08T14:00:46.581+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO DAGScheduler: Got job 0 (load at NativeMethodAccessorImpl.java:0) with 1 output partitions
[2024-12-08T14:00:46.583+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO DAGScheduler: Final stage: ResultStage 0 (load at NativeMethodAccessorImpl.java:0)
[2024-12-08T14:00:46.583+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO DAGScheduler: Parents of final stage: List()
[2024-12-08T14:00:46.585+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO DAGScheduler: Missing parents: List()
[2024-12-08T14:00:46.610+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO DAGScheduler: Submitting ResultStage 0 (MapPartitionsRDD[3] at load at NativeMethodAccessorImpl.java:0), which has no missing parents
[2024-12-08T14:00:46.821+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO MemoryStore: Block broadcast_1 stored as values in memory (estimated size 13.5 KiB, free 434.2 MiB)
[2024-12-08T14:00:46.823+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO MemoryStore: Block broadcast_1_piece0 stored as bytes in memory (estimated size 6.4 KiB, free 434.2 MiB)
[2024-12-08T14:00:46.824+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO BlockManagerInfo: Added broadcast_1_piece0 in memory on 2a892a3469ec:33843 (size: 6.4 KiB, free: 434.4 MiB)
[2024-12-08T14:00:46.826+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO SparkContext: Created broadcast 1 from broadcast at DAGScheduler.scala:1585
[2024-12-08T14:00:46.870+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO DAGScheduler: Submitting 1 missing tasks from ResultStage 0 (MapPartitionsRDD[3] at load at NativeMethodAccessorImpl.java:0) (first 15 tasks are for partitions Vector(0))
[2024-12-08T14:00:46.874+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:46 INFO TaskSchedulerImpl: Adding task set 0.0 with 1 tasks resource profile 0
[2024-12-08T14:00:53.635+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:53 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20241208140035-0003/0 on worker-20241208140008-172.18.0.6-34549 (172.18.0.6:34549) with 2 core(s)
[2024-12-08T14:00:53.638+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:53 INFO StandaloneSchedulerBackend: Granted executor ID app-20241208140035-0003/0 on hostPort 172.18.0.6:34549 with 2 core(s), 1024.0 MiB RAM
[2024-12-08T14:00:53.696+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:53 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20241208140035-0003/0 is now RUNNING
[2024-12-08T14:00:56.477+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:56 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Registered executor NettyRpcEndpointRef(spark-client://Executor) (172.18.0.6:35524) with ID 0,  ResourceProfileId 0
[2024-12-08T14:00:56.526+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:56 INFO BlockManagerMasterEndpoint: Registering block manager 172.18.0.6:37297 with 434.4 MiB RAM, BlockManagerId(0, 172.18.0.6, 37297, None)
[2024-12-08T14:00:56.577+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:56 INFO TaskSetManager: Starting task 0.0 in stage 0.0 (TID 0) (172.18.0.6, executor 0, partition 0, PROCESS_LOCAL, 9852 bytes)
[2024-12-08T14:00:56.892+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:56 INFO BlockManagerInfo: Added broadcast_1_piece0 in memory on 172.18.0.6:37297 (size: 6.4 KiB, free: 434.4 MiB)
[2024-12-08T14:00:57.470+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:57 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 172.18.0.6:37297 (size: 34.3 KiB, free: 434.4 MiB)
[2024-12-08T14:00:57.803+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:57 WARN TaskSetManager: Lost task 0.0 in stage 0.0 (TID 0) (172.18.0.6 executor 0): org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-1cd0e7b8-9745-43c8-8c3a-c004433dfa05/userFiles-288d223e-9a2c-48e5-acd4-691d23aa6c92/test1234.csv does not exist
[2024-12-08T14:00:57.803+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-08T14:00:57.804+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-08T14:00:57.804+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-08T14:00:57.804+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-08T14:00:57.804+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-08T14:00:57.804+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-08T14:00:57.805+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-08T14:00:57.805+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-08T14:00:57.805+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-08T14:00:57.805+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-08T14:00:57.806+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-08T14:00:57.806+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-08T14:00:57.806+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-08T14:00:57.807+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-08T14:00:57.807+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-08T14:00:57.807+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-08T14:00:57.808+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-08T14:00:57.808+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-08T14:00:57.808+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-08T14:00:57.809+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-08T14:00:57.809+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-08T14:00:57.809+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-08T14:00:57.809+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-08T14:00:57.810+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-08T14:00:57.810+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-08T14:00:57.811+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-08T14:00:57.811+0000] {spark_submit.py:634} INFO - 
[2024-12-08T14:00:57.812+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:57 INFO TaskSetManager: Starting task 0.1 in stage 0.0 (TID 1) (172.18.0.6, executor 0, partition 0, PROCESS_LOCAL, 9852 bytes)
[2024-12-08T14:00:57.846+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:57 INFO TaskSetManager: Lost task 0.1 in stage 0.0 (TID 1) on 172.18.0.6, executor 0: org.apache.spark.SparkFileNotFoundException (File file:/tmp/spark-1cd0e7b8-9745-43c8-8c3a-c004433dfa05/userFiles-288d223e-9a2c-48e5-acd4-691d23aa6c92/test1234.csv does not exist
[2024-12-08T14:00:57.847+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.) [duplicate 1]
[2024-12-08T14:00:57.848+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:57 INFO TaskSetManager: Starting task 0.2 in stage 0.0 (TID 2) (172.18.0.6, executor 0, partition 0, PROCESS_LOCAL, 9852 bytes)
[2024-12-08T14:00:57.883+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:57 INFO TaskSetManager: Lost task 0.2 in stage 0.0 (TID 2) on 172.18.0.6, executor 0: org.apache.spark.SparkFileNotFoundException (File file:/tmp/spark-1cd0e7b8-9745-43c8-8c3a-c004433dfa05/userFiles-288d223e-9a2c-48e5-acd4-691d23aa6c92/test1234.csv does not exist
[2024-12-08T14:00:57.889+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.) [duplicate 2]
[2024-12-08T14:00:57.890+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:57 INFO TaskSetManager: Starting task 0.3 in stage 0.0 (TID 3) (172.18.0.6, executor 0, partition 0, PROCESS_LOCAL, 9852 bytes)
[2024-12-08T14:00:57.965+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:57 INFO TaskSetManager: Lost task 0.3 in stage 0.0 (TID 3) on 172.18.0.6, executor 0: org.apache.spark.SparkFileNotFoundException (File file:/tmp/spark-1cd0e7b8-9745-43c8-8c3a-c004433dfa05/userFiles-288d223e-9a2c-48e5-acd4-691d23aa6c92/test1234.csv does not exist
[2024-12-08T14:00:57.967+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.) [duplicate 3]
[2024-12-08T14:00:57.968+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:57 ERROR TaskSetManager: Task 0 in stage 0.0 failed 4 times; aborting job
[2024-12-08T14:00:57.972+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:57 INFO TaskSchedulerImpl: Removed TaskSet 0.0, whose tasks have all completed, from pool
[2024-12-08T14:00:57.978+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:57 INFO TaskSchedulerImpl: Cancelling stage 0
[2024-12-08T14:00:57.978+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:57 INFO TaskSchedulerImpl: Killing all running tasks in stage 0: Stage cancelled: Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3) (172.18.0.6 executor 0): org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-1cd0e7b8-9745-43c8-8c3a-c004433dfa05/userFiles-288d223e-9a2c-48e5-acd4-691d23aa6c92/test1234.csv does not exist
[2024-12-08T14:00:57.980+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-08T14:00:57.981+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-08T14:00:57.982+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-08T14:00:57.982+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-08T14:00:57.983+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-08T14:00:57.984+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-08T14:00:57.984+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-08T14:00:57.985+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-08T14:00:57.986+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-08T14:00:57.987+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-08T14:00:57.987+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-08T14:00:57.988+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-08T14:00:57.989+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-08T14:00:57.989+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-08T14:00:57.991+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-08T14:00:57.992+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-08T14:00:57.993+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-08T14:00:57.993+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-08T14:00:57.994+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-08T14:00:57.994+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-08T14:00:57.995+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-08T14:00:57.995+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-08T14:00:57.996+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-08T14:00:57.997+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-08T14:00:57.997+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-08T14:00:57.998+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-08T14:00:57.999+0000] {spark_submit.py:634} INFO - 
[2024-12-08T14:00:57.999+0000] {spark_submit.py:634} INFO - Driver stacktrace:
[2024-12-08T14:00:58.000+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:57 INFO DAGScheduler: ResultStage 0 (load at NativeMethodAccessorImpl.java:0) failed in 11.346 s due to Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3) (172.18.0.6 executor 0): org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-1cd0e7b8-9745-43c8-8c3a-c004433dfa05/userFiles-288d223e-9a2c-48e5-acd4-691d23aa6c92/test1234.csv does not exist
[2024-12-08T14:00:58.001+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-08T14:00:58.001+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-08T14:00:58.002+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-08T14:00:58.003+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-08T14:00:58.004+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-08T14:00:58.005+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-08T14:00:58.006+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-08T14:00:58.007+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-08T14:00:58.008+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-08T14:00:58.009+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-08T14:00:58.009+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-08T14:00:58.010+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-08T14:00:58.011+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-08T14:00:58.013+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-08T14:00:58.014+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-08T14:00:58.014+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-08T14:00:58.015+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-08T14:00:58.017+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-08T14:00:58.017+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-08T14:00:58.018+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-08T14:00:58.018+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-08T14:00:58.018+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-08T14:00:58.019+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-08T14:00:58.019+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-08T14:00:58.019+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-08T14:00:58.020+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-08T14:00:58.020+0000] {spark_submit.py:634} INFO - 
[2024-12-08T14:00:58.021+0000] {spark_submit.py:634} INFO - Driver stacktrace:
[2024-12-08T14:00:58.021+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:57 INFO DAGScheduler: Job 0 failed: load at NativeMethodAccessorImpl.java:0, took 11.438558 s
[2024-12-08T14:00:58.399+0000] {spark_submit.py:634} INFO - Traceback (most recent call last):
[2024-12-08T14:00:58.399+0000] {spark_submit.py:634} INFO - File "/opt/***/jobs/python/wordcountjob.py", line 54, in <module>
[2024-12-08T14:00:58.401+0000] {spark_submit.py:634} INFO - df = spark.read.load(file_path, format="csv", inferSchema="true", header="true")
[2024-12-08T14:00:58.402+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-08T14:00:58.402+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/readwriter.py", line 307, in load
[2024-12-08T14:00:58.402+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2024-12-08T14:00:58.403+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 179, in deco
[2024-12-08T14:00:58.404+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/protocol.py", line 326, in get_return_value
[2024-12-08T14:00:58.445+0000] {spark_submit.py:634} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o28.load.
[2024-12-08T14:00:58.445+0000] {spark_submit.py:634} INFO - : org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3) (172.18.0.6 executor 0): org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-1cd0e7b8-9745-43c8-8c3a-c004433dfa05/userFiles-288d223e-9a2c-48e5-acd4-691d23aa6c92/test1234.csv does not exist
[2024-12-08T14:00:58.445+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-08T14:00:58.446+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-08T14:00:58.446+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-08T14:00:58.447+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-08T14:00:58.447+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-08T14:00:58.448+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-08T14:00:58.448+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-08T14:00:58.448+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-08T14:00:58.448+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-08T14:00:58.448+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-08T14:00:58.451+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-08T14:00:58.452+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-08T14:00:58.453+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-08T14:00:58.453+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-08T14:00:58.454+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-08T14:00:58.454+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-08T14:00:58.455+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-08T14:00:58.455+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-08T14:00:58.455+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-08T14:00:58.456+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-08T14:00:58.456+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-08T14:00:58.456+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-08T14:00:58.456+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-08T14:00:58.456+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-08T14:00:58.457+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-08T14:00:58.457+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-08T14:00:58.457+0000] {spark_submit.py:634} INFO - 
[2024-12-08T14:00:58.457+0000] {spark_submit.py:634} INFO - Driver stacktrace:
[2024-12-08T14:00:58.458+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2856)
[2024-12-08T14:00:58.458+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2792)
[2024-12-08T14:00:58.458+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2791)
[2024-12-08T14:00:58.458+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2024-12-08T14:00:58.459+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2024-12-08T14:00:58.459+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2024-12-08T14:00:58.459+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2791)
[2024-12-08T14:00:58.459+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1247)
[2024-12-08T14:00:58.459+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1247)
[2024-12-08T14:00:58.459+0000] {spark_submit.py:634} INFO - at scala.Option.foreach(Option.scala:407)
[2024-12-08T14:00:58.460+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1247)
[2024-12-08T14:00:58.460+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:3060)
[2024-12-08T14:00:58.460+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2994)
[2024-12-08T14:00:58.460+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2983)
[2024-12-08T14:00:58.460+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
[2024-12-08T14:00:58.460+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:989)
[2024-12-08T14:00:58.461+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.runJob(SparkContext.scala:2393)
[2024-12-08T14:00:58.461+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.runJob(SparkContext.scala:2414)
[2024-12-08T14:00:58.461+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.runJob(SparkContext.scala:2433)
[2024-12-08T14:00:58.461+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:530)
[2024-12-08T14:00:58.461+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:483)
[2024-12-08T14:00:58.462+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.CollectLimitExec.executeCollect(limit.scala:61)
[2024-12-08T14:00:58.462+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.collectFromPlan(Dataset.scala:4333)
[2024-12-08T14:00:58.462+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.$anonfun$head$1(Dataset.scala:3316)
[2024-12-08T14:00:58.462+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.$anonfun$withAction$2(Dataset.scala:4323)
[2024-12-08T14:00:58.462+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:546)
[2024-12-08T14:00:58.463+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:4321)
[2024-12-08T14:00:58.463+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:125)
[2024-12-08T14:00:58.463+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:201)
[2024-12-08T14:00:58.463+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:108)
[2024-12-08T14:00:58.463+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:900)
[2024-12-08T14:00:58.463+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:66)
[2024-12-08T14:00:58.463+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.withAction(Dataset.scala:4321)
[2024-12-08T14:00:58.464+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.head(Dataset.scala:3316)
[2024-12-08T14:00:58.464+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.take(Dataset.scala:3539)
[2024-12-08T14:00:58.464+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.csv.TextInputCSVDataSource$.infer(CSVDataSource.scala:111)
[2024-12-08T14:00:58.464+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.csv.CSVDataSource.inferSchema(CSVDataSource.scala:64)
[2024-12-08T14:00:58.464+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.csv.CSVFileFormat.inferSchema(CSVFileFormat.scala:62)
[2024-12-08T14:00:58.465+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.$anonfun$getOrInferFileFormatSchema$11(DataSource.scala:208)
[2024-12-08T14:00:58.465+0000] {spark_submit.py:634} INFO - at scala.Option.orElse(Option.scala:447)
[2024-12-08T14:00:58.465+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.getOrInferFileFormatSchema(DataSource.scala:205)
[2024-12-08T14:00:58.465+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:407)
[2024-12-08T14:00:58.466+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:229)
[2024-12-08T14:00:58.466+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:211)
[2024-12-08T14:00:58.466+0000] {spark_submit.py:634} INFO - at scala.Option.getOrElse(Option.scala:189)
[2024-12-08T14:00:58.466+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:211)
[2024-12-08T14:00:58.466+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:186)
[2024-12-08T14:00:58.467+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2024-12-08T14:00:58.467+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2024-12-08T14:00:58.467+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-12-08T14:00:58.467+0000] {spark_submit.py:634} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2024-12-08T14:00:58.467+0000] {spark_submit.py:634} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2024-12-08T14:00:58.468+0000] {spark_submit.py:634} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-12-08T14:00:58.468+0000] {spark_submit.py:634} INFO - at py4j.Gateway.invoke(Gateway.java:282)
[2024-12-08T14:00:58.468+0000] {spark_submit.py:634} INFO - at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2024-12-08T14:00:58.468+0000] {spark_submit.py:634} INFO - at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2024-12-08T14:00:58.469+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-12-08T14:00:58.469+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-12-08T14:00:58.469+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-08T14:00:58.469+0000] {spark_submit.py:634} INFO - Caused by: org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-1cd0e7b8-9745-43c8-8c3a-c004433dfa05/userFiles-288d223e-9a2c-48e5-acd4-691d23aa6c92/test1234.csv does not exist
[2024-12-08T14:00:58.469+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-08T14:00:58.469+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-08T14:00:58.470+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-08T14:00:58.470+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-08T14:00:58.470+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-08T14:00:58.470+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-08T14:00:58.470+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-08T14:00:58.471+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-08T14:00:58.471+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-08T14:00:58.471+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-08T14:00:58.471+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-08T14:00:58.472+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-08T14:00:58.472+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-08T14:00:58.473+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-08T14:00:58.473+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-08T14:00:58.473+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-08T14:00:58.473+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-08T14:00:58.473+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-08T14:00:58.473+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-08T14:00:58.473+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-08T14:00:58.474+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-08T14:00:58.474+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-08T14:00:58.475+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-08T14:00:58.475+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-08T14:00:58.475+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-08T14:00:58.475+0000] {spark_submit.py:634} INFO - ... 1 more
[2024-12-08T14:00:58.475+0000] {spark_submit.py:634} INFO - 
[2024-12-08T14:00:58.549+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:58 INFO SparkContext: Invoking stop() from shutdown hook
[2024-12-08T14:00:58.550+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:58 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2024-12-08T14:00:58.559+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:58 INFO SparkUI: Stopped Spark web UI at http://2a892a3469ec:4045
[2024-12-08T14:00:58.562+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:58 INFO StandaloneSchedulerBackend: Shutting down all executors
[2024-12-08T14:00:58.562+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:58 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2024-12-08T14:00:58.587+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:58 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2024-12-08T14:00:58.651+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:58 INFO MemoryStore: MemoryStore cleared
[2024-12-08T14:00:58.654+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:58 INFO BlockManager: BlockManager stopped
[2024-12-08T14:00:58.676+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:58 INFO BlockManagerMaster: BlockManagerMaster stopped
[2024-12-08T14:00:58.682+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:58 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2024-12-08T14:00:58.708+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:58 INFO SparkContext: Successfully stopped SparkContext
[2024-12-08T14:00:58.709+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:58 INFO ShutdownHookManager: Shutdown hook called
[2024-12-08T14:00:58.711+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:58 INFO ShutdownHookManager: Deleting directory /tmp/spark-1cd0e7b8-9745-43c8-8c3a-c004433dfa05/pyspark-6e803554-11a2-45b3-9b41-e54163777b01
[2024-12-08T14:00:58.730+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:58 INFO ShutdownHookManager: Deleting directory /tmp/spark-1cd0e7b8-9745-43c8-8c3a-c004433dfa05
[2024-12-08T14:00:58.737+0000] {spark_submit.py:634} INFO - 24/12/08 14:00:58 INFO ShutdownHookManager: Deleting directory /tmp/spark-c9e693a9-d7fd-4635-a295-40138ebd2409
[2024-12-08T14:00:58.815+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.
[2024-12-08T14:00:58.826+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=sparking_flow, task_id=python_job, run_id=scheduled__2024-12-06T00:00:00+00:00, execution_date=20241206T000000, start_date=20241208T140021, end_date=20241208T140058
[2024-12-08T14:00:58.856+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-12-08T14:00:58.858+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 135 for task python_job (Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.; 197)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.
[2024-12-08T14:00:58.901+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-12-08T14:00:58.957+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-12-08T14:00:58.995+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
